modules/common/src/main/scala/com.snowplowanalytics.snowplow.enrich/common/enrichments/web/PageEnrichments.scala

```diff
@@ -52,19 +53,12 @@ object PageEnrichments {
    * @param qsMap The querystring parameters
    * @return Validation boxing a pair of optional strings corresponding to the two fields
    */
-  def parseCrossDomain(qsMap: QueryStringParameters): Either[FailureDetails.EnrichmentFailure, (Option[String], Option[String])] =
+  def parseCrossDomain(qsMap: QueryStringParameters): Either[FailureDetails.EnrichmentFailure, Map[String, Option[String]]] =
```

Comment:
I had some back and forths on this. Happy to move this over completely, it just felt that this function parses querystring params and the fact that it deals only with `_sp` is "coincidence", e.g. maybe in the future it deals with more. What do you think?

Reply:
Ah I see your point, but in case we have more query parameters to parse, wouldn't we have multiple functions for parsing each one separately? Not sure I see the value of having it in one, but this is just a nitpick, this approach sounds fine as well.